extensions/common/json-ld/src/main/java/org/eclipse/edc/jsonld/transformer/from/JsonObjectFromQuerySpecTransformer.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.jsonld.transformer.from;
+
+import jakarta.json.JsonArrayBuilder;
+import jakarta.json.JsonBuilderFactory;
+import jakarta.json.JsonObject;
+import org.eclipse.edc.jsonld.spi.transformer.AbstractJsonLdTransformer;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.edc.transform.spi.TransformerContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static java.util.UUID.randomUUID;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.ID;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.TYPE;
+
+public class JsonObjectFromQuerySpecTransformer extends AbstractJsonLdTransformer<QuerySpec, JsonObject> {
+    private final JsonBuilderFactory jsonFactory;
+
+    public JsonObjectFromQuerySpecTransformer(JsonBuilderFactory jsonFactory) {
+        super(QuerySpec.class, JsonObject.class);
+        this.jsonFactory = jsonFactory;
+    }
+
+    @Override
+    public @Nullable JsonObject transform(@NotNull QuerySpec querySpec, @NotNull TransformerContext context) {
+        var builder = jsonFactory.createObjectBuilder();
+        builder.add(ID, randomUUID().toString());
+        builder.add(TYPE, QuerySpec.EDC_QUERY_SPEC_TYPE);
+        builder.add(QuerySpec.EDC_QUERY_SPEC_LIMIT, querySpec.getLimit());
+        builder.add(QuerySpec.EDC_QUERY_SPEC_OFFSET, querySpec.getOffset());
+        builder.add(QuerySpec.EDC_QUERY_SPEC_SORT_ORDER, querySpec.getSortOrder().toString());
+
+        if (querySpec.getSortField() != null) {
+            builder.add(QuerySpec.EDC_QUERY_SPEC_SORT_FIELD, querySpec.getSortField());
+        }
+
+        var filterExpressions = querySpec.getFilterExpression().stream()
+                .map(distribution -> context.transform(distribution, JsonObject.class))
```

Comment:
handle error

Reply:
That should be handled by the inner transformer to report the error